Increase cooldown timer and adjust in-game timer.

Ovalelephant35 opened the pull request:
Increase cooldown timer and adjust in-game timer
* Cooldown timer between levels increased from 3 to 6 seconds.
* In-game timer now increases by 2 seconds for each level.

There is need for more time in between levels to read the instruction for what has to be done in next levels, so increased cooldown timer.
With every level increase , I increased the in-game timer by 2 seconds to keep game fair.
Thankyou.

Kindly review @quozl @walterbender @chimosky .

Commit:
Increase cooldown timer and adjust in-game timer

* Cooldown timer between levels increased from 3 to 6 seconds.

* In-game timer now increases by 2 seconds for each level.

chimosky commented on gamearea.py:
```diff
@@ -650,7 +650,7 @@ def cb():
         self.playing = True
 
-        self.start_timeout(15, cb)
+        self.start_timeout(15 + 2 * (self.level-2), cb)
```
What's the reasoning here?

Ovalelephant35 replied:
Corrected the level initialization see this one first https://github.com/sugarlabs/classify-cats/pull/12.
Corrected the logic here also as 15 + 2*(self.level-1);
logic is base time is 15 sec:
then for level 1 = 15+2*(1-1) = 15;
then for level 2 = 15+2*(2-1) = 17; and so on.

chimosky replied:
If a problem occurred here, why open another PR to fix it?

Ovalelephant35 replied:
I got confused as previously I though game was starting at level one thats why I used - 2 as control but then made another branch to test and see whats exactly happening it is related to how level.json file is being accessed and also both of them are unrelated one is bug other is feature though could have been made in one.

Commit:
Fix bug causing game level to start at level 2 instead of level 1

* The "start()" method incorrectly set "self.win = True",
  causing "get_next_level()" to increment the level by 1,
  resulting in the game starting from level 2.

* "self.win"= False` in the `start()` method,
  ensuring the game starts at the correct level.

* Above Changes caused Wrong score for level 1 as 0,
  Implemented the score = 20 which is only possible score
  achieved as there is no purpose of reaction time in this level,
  also taken care of edge case when someone exhaust all
  the attempts at level 1.

ghost commented:
Instead of going on with self.level = 0 and defining two states as current as next state, I found out the reason that from the point a person was hitting the star button to start the game, it was being considered as next state because of self.win initialised to true I changed that.

ghost commented:
> Instead of going on with self.level = 0 and defining two states as current as next state, I found out the reason that from the point a person was hitting the star button to start the game, it was being considered as next state because of self.win initialised to true I changed that.

But there is a error associated with it and that is for the first iteration your score come out to be 0 even if you win, and writing condition as 
`if self.win or self.level == 1` would not resolve as if you lose it keep accumulating points so hardcoding points and taking care of edge case option which is
``` 
def generate_score(self, reaction_time):
        score = self.score
        if self.level == 1 and self.max_puzzle_count == 10 and self.win is False:
            score = 0
        if self.level == 1:
            score = 20
        if self.win:
            score = self.score + reaction_time + 20
        return score
```
I am making self.max_puzzle_count == 10 so you can merge https://github.com/sugarlabs/classify-cats/pull/14/ and test it.

chimosky commented on gamearea.py:
```diff
@@ -624,6 +624,10 @@ def get_next_level(self):
         score = self.score
+        if self.level == 1 and self.max_puzzle_count == 10 and self.win is False:
+            score = 0
+        if self.level == 1:
+            score = 20
```
The score should only increase when a level is cleared and not because of a certain level.

Ovalelephant35 replied:
You will only get to know your score if you clear the level, it is acting as a storage thats it.
There are two ways to come out of game either you press star which will lead to new game or exhaust all the attempts which is take care of.

Ovalelephant35 replied:
And if you lose it will tell you to try again until you clear the level or exhaust all the attempts.

chimosky replied:
Say a player fails at level one, that check automatically makes their score 20 when they failed.

chimosky commented:
> No there is get level that handling what level has to be selected and that is with the help of self.level whose initialisation has to be done to 0 as level = self.level+1. so that game progresses it is there to help in initialisation.
>
> So basically initialization of self.level is like current state of the game which has to start at 0 so that level which will how the game will play out can be self.level+1.

Responding to this here as you've moved the convo here, initialization of level isn't `self.level + 1`, that assignment only happens if self.win, the game seems to start out with self.win being `True` when it shouldn't be as it should only be true when a level is cleared.

> But there is a error associated with it and that is for the first iteration your score come out to be 0 even if you win, and writing condition as
if self.win or self.level == 1 would not resolve as if you lose it keep accumulating points so hardcoding points and taking care of edge case option which is

Figure out why the score isn't been calculated at level 1 even if you win, incrementing the score at level 1 regardless of the outcome doesn't solve the problem.

> I am making self.max_puzzle_count == 10 so you can merge https://github.com/sugarlabs/classify-cats/pull/14 and test it.

When working on a PR, if it's not related to another PR then refrain from mentioning it.
If the change was caused by something you noticed here, then fix it here instead of opening another PR.

ghost commented:
No, the logic of game is designed that way, self.win = true is helping in moving forward towards other level.
Regarding increasing the score regardless of win or loss, it does solve the problem see the whole code once, you will only get to know your score when you clear the level, if not it will tell you to try again see the functions and code once all of it, how the messages are being displayed etc.
Now one might think what if I exhaust all my attempts at level one only that is also covered see the edge case.

All of logic is already embedded in the game.

And regarding mentioning other PR because I used the edge case as 10 not 5 thats why mentioned it otherwise more confusion will arise.

ghost commented:
Basically we have two solution to move forward with :-
1- Either make self.level = 0 and assume that before pressing the star, game has already started which is the current state with self.win = true, as I mentioned earlier with next state being called upon by get.level.
2- Other one is to assume that game is going to start after pressing the star button at level 1 with self.win = false but somehow self.win is not being updated in time.

Last way is to change all the logic.

chimosky commented:
> No, the logic of game is designed that way, self.win = true is helping in moving forward towards other level. Regarding increasing the score regardless of win or loss, it does solve the problem see the whole code once, you will only get to know your score when you clear the level, if not it will tell you to try again see the functions and code once all of it, how the messages are being displayed etc. Now one might think what if I exhaust all my attempts at level one only that is also covered see the edge case.
>

I was talking about `self.win = True` when the game is just starting, although I agree with not changing that logic at the moment.

Again, incrementing the score regardless doesn't solve any problem, the code for generating score does it's job perfectly.

ghost commented:
Though it does solve the problem, but as mentioned by seeing it seems to have  introduced some more problems thanks closing.
